sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -593,16 +593,27 @@ protected EventProcessor()
 
             if ((diagnosticScope.IsEnabled) && (eventBatch.Any()))
             {
+                bool isBatch = (eventBatch.Count > 1);
                 foreach (var eventData in eventBatch)
                 {
                     if (EventDataInstrumentation.TryExtractDiagnosticId(eventData, out string diagnosticId))
                     {
-                        var attributes = new Dictionary<string, string>(1)
+                        if (isBatch)
                         {
-                            { DiagnosticProperty.EnqueuedTimeAttribute, eventData.EnqueuedTime.ToUnixTimeMilliseconds().ToString(CultureInfo.InvariantCulture) }
-                        };
-
-                        diagnosticScope.AddLink(diagnosticId, null, attributes);
+                            var attributes = new Dictionary<string, string>(1)
+                            {
```

Comment:
I wonder if this is really the right approach, and I'd like to talk things through and better understand the intent before this moves forward.

Events are should always be read in batches, even when they're being dispatched for processing individually.  Under normal circumstances, if a processor is reading a single event, it's a sign that the prefetch is not tuned properly or there are network issues.

We do not recommend using a batch size of 1, outside of some very unusual edge cases, as it would have a large performance impact.

Reply:
One scenario where this will come into play is using the Event Hub trigger for a function that receives a single event.

Reply:
I guess I see your point if it's possible that some events are delivered in batches and some are not in the lifetime of a single EventProcessor. If that could happen, this would probably cause more confusion than help. @lmolkova - what do you think?

Reply:
I agree that being consistent is good, i.e. if as a user I sometimes receive batches, it's best to have the same tracing experience when I accidentally receive a single message.

But if I only receive a single message (either with EventHubPoricessor or Function), having parent-child relationship is better.

So, if `MaxBatchSize` is the setting that controls it, I agree we should use it to control the behavior and not the actual batch size.

Reply:
BTW, in any case there should not be any problem on Azure Monitor with either of these. AzMon supports both cases just fine and Track1 SDKs used parent-child relationships.

Reply:
Josh and I chatted, and we're going to keep the basic structure and just key off of the requested batch size.   That'll keep things consistent per-read.